uefi-test-runner/src/proto/device_path.rs

```diff
@@ -7,57 +9,94 @@ use uefi::table::boot::BootServices;
+
+        for path in device_path.node_iter() {
+            info!(
+                "path: type={:?}, subtype={:?}, length={}",
+                path.device_type(),
+                path.sub_type(),
+                path.length(),
+            );
 
-        let text = device_path_to_text
-            .convert_device_node_to_text(bt, path, DisplayOnly(true), AllowShortcuts(false))
-            .expect("Failed to convert device path to text");
-        let text = &*text;
-        info!("path name: {text}");
+            let text = device_path_to_text
+                .convert_device_node_to_text(bt, path, DisplayOnly(true), AllowShortcuts(false))
+                .expect("Failed to convert device path to text");
+            let text = &*text;
+            info!("path name: {text}");
 
-        let convert = device_path_from_text
-            .convert_text_to_device_node(text)
-            .expect("Failed to convert text to device path");
-        assert_eq!(path, convert);
+            let convert = device_path_from_text
+                .convert_text_to_device_node(text)
+                .expect("Failed to convert text to device path");
+            assert_eq!(path, convert);
+        }
+
+        // Get the `LoadedImageDevicePath`. Verify it start with the same nodes as
+        // `device_path`.
+        let loaded_image_device_path = bt
+            .open_protocol_exclusive::<LoadedImageDevicePath>(image)
+            .expect("Failed to open LoadedImageDevicePath protocol");
+
+        for (n1, n2) in device_path
+            .node_iter()
+            .zip(loaded_image_device_path.node_iter())
+        {
+            assert_eq!(n1, n2);
+        }
     }
 
-    // Get the `LoadedImageDevicePath`. Verify it start with the same nodes as
-    // `device_path`.
-    let loaded_image_device_path = bt
-        .open_protocol_exclusive::<LoadedImageDevicePath>(image)
-        .expect("Failed to open LoadedImageDevicePath protocol");
-    for (n1, n2) in device_path
-        .node_iter()
-        .zip(loaded_image_device_path.node_iter())
+    // test 2/2: test high-level to-string api
     {
-        assert_eq!(n1, n2);
+        let loaded_image_device_path = bt
+            .open_protocol_exclusive::<LoadedImageDevicePath>(image)
+            .expect("Failed to open LoadedImageDevicePath protocol");
+        let device_path: &DevicePath = &loaded_image_device_path;
+
+        let path = device_path
+            .node_iter()
+            .map(|node| node.to_string(bt, DisplayOnly(false), AllowShortcuts(false)))
+            .map(|str| str.unwrap().unwrap().to_string())
+            .collect::<Vec<_>>();
+
+        assert_eq!(
+            path.as_slice(),
+            &[
+                "PciRoot(0x0)",
+                "Pci(0x1F,0x2)",
+                "Sata(0x0,0xFFFF,0x0)",
+                "HD(1,MBR,0xBE1AFDFA,0x3F,0xFBFC1)",
+                "\\efi\\boot\\test_runner.efi"
+            ]
+        );
```

Comment:
The path will be different on aarch64, so might want to use an `if cfg!(...)` here

Reply:
right! At the first glance, I do not see where this difference comes from. Can you give me a pointer, please? I guess, its the way that xtask prepares the QEMU command.